EssentialCSharp.Web/Services/Referrals/ReferralService.cs

```diff
@@ -0,0 +1,117 @@
+using System.Security.Claims;
+using EssentialCSharp.Web.Areas.Identity.Data;
+using EssentialCSharp.Web.Data;
+using Microsoft.AspNetCore.Identity;
+using Microsoft.EntityFrameworkCore;
+using Sqids;
+
+namespace EssentialCSharp.Web.Services.Referrals;
+
+public class ReferralService(EssentialCSharpWebContext dbContext, SqidsEncoder<int> sqids, UserManager<EssentialCSharpWebUser> userManager) : IReferralService
+{
+    public async Task<string?> GetReferralIdAsync(string userId)
+    {
+        EssentialCSharpWebUser? user = await userManager.FindByIdAsync(userId);
+        return await GetReferralIdAsync(user);
+    }
+
+    public async Task<string?> GetReferralIdAsync(EssentialCSharpWebUser? user)
+    {
+        if (user is null)
+        {
+            return null;
+        }
+        else
+        {
+            // Check if the user already has a referrer ID
+            if (!string.IsNullOrEmpty(user.ReferrerId))
+            {
+                return user.ReferrerId;
+            }
+            else
+            {
+                Random random = Random.Shared;
+                string referrerId = sqids.Encode(random.Next());
+                user.ReferrerId = referrerId;
```

Comment:
Not sure I understand the point of encoding a random number like this. The number is already random and therefore more or less unguessable - another layer on top of that just introduces complexity. If you don't want this to be numeric, then don't use a number - generate some random bytes and base64 encode them, then store that string in the database.

Reply:
This is mostly due to just wanting to try a new library and I like the way the id's look, nothing more than that :) I could base64 encode bytes instead, but of course then I'd have to strip out +, / and = (obviously easy enough) for url compatibility, but I think for now I like trying this new library since it changed from Hashids

Reply:
No worries, that sounds like a stronger opinion that I have :) Changed in favor of encoded bytes